### ISSUE TITLE
clone issue in 4.0.0

### ISSUE THREAD

**Issue by adamgreenhall** [conda/conda#2178]
getting this error with ``conda==4.0.0``. ``conda==3.19.3`` works just fine.

```
$ conda create -y -n shiny_new_env --clone root
Fetching package metadata: ....
src_prefix: '/srv/conda'
dst_prefix: '/srv/conda/envs/shiny_new_env'
Packages: 97
Files: 6350
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.

Traceback (most recent call last):
  File "/srv/conda/bin/conda", line 6, in <module>
    sys.exit(main())
  File "/srv/conda/lib/python2.7/site-packages/conda/cli/main.py", line 139, in main
    args_func(args, p)
  File "/srv/conda/lib/python2.7/site-packages/conda/cli/main.py", line 146, in args_func
    args.func(args, p)
  File "/srv/conda/lib/python2.7/site-packages/conda/cli/main_create.py", line 49, in execute
    install.install(args, parser, 'create')
  File "/srv/conda/lib/python2.7/site-packages/conda/cli/install.py", line 247, in install
    clone(args.clone, prefix, json=args.json, quiet=args.quiet, index=index)
  File "/srv/conda/lib/python2.7/site-packages/conda/cli/install.py", line 84, in clone
    quiet=quiet, index=index)
  File "/srv/conda/lib/python2.7/site-packages/conda/misc.py", line 243, in clone_env
    sorted_dists = r.dependency_sort(dists)
  File "/srv/conda/lib/python2.7/site-packages/conda/resolve.py", line 716, in dependency_sort
    for key, value in iteritems(must_have):
  File "/srv/conda/lib/python2.7/site-packages/conda/compat.py", line 157, in iteritems
    return iter(getattr(d, _iteritems)())
AttributeError: 'list' object has no attribute 'iteritems'

```

```
$ conda info
Current conda install:

             platform : linux-64
        conda version : 4.0.0
  conda-build version : not installed
       python version : 2.7.11.final.0
     requests version : 2.5.0
     root environment : /srv/conda  (read only)
  default environment : /srv/conda
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False
```

**Comment by ilanschnell** [comment]
CC @kalefranz

**Comment by ilanschnell** [comment]
Thanks for posting this issue.  The conda 4.0 release was a big deal, and brought many changes with it.  Even though we have done more the usual testing, and two public beta releases, we missed this issue.  We will probably release conda 4.0.1 on Monday or Tuesday.

**Comment by mcg1969** [comment]
Looks like it is the same issue as https://github.com/conda/conda/issues/2160, which was closed because it seemed like updating conda fixed it. But it looks like there are some circumstances where it is still an issue. @ilanschnell, I think the problem is that `resolve.dependency_sort` is expecting a dictionary of packages---package names as keys, dists as values---but the `clone` function seems to be sending it a list instead.

**Comment by kalefranz** [comment]
This stack trace has been fixed in `4.0.1`.

**Comment by adamgreenhall** [comment]
thanks!

**Comment by mcg1969** [comment]
Fix in preparation.

**Comment by mcg1969** [comment]
Fix here: https://github.com/conda/conda/pull/2193

**Comment by edurand** [comment]
Thanks! When will this be released?

**Comment by mcg1969** [comment]
4.0.2 I assume, do not know the timeline

**Comment by kalefranz** [comment]
Should be fixed with today's `4.0.2` release.
